### PR TITLE
Feature/d3 asim 3669: Rebranding

### PR DESCRIPTION
**Pull request by faizan2590**

### CODE REVIEW

**Review by hannesdiedrich — DISMISSED** [review]
LGTM

**Review by Hassan754 — APPROVED** [review]
LGTM

**Review by spyrostz — DISMISSED** [review]
LGTM

**Review by spyrostz — APPROVED** [review]
LGTM

File: d3a_interface/__init__.py [hunk 1]
```diff
@@ -0,0 +1,5 @@
+import sys
+
+import gsy_framework  # NOQA
+
+sys.modules[__name__] = sys.modules['gsy_framework']
```
Comment (fievelk) [inline d3a_interface/__init__.py:5]: Could you explain what this code is for? Will users be able to still use `import d3a_interface`?

Nit:
```suggestion
import gsy_framework

sys.modules[__name__] = sys.modules["gsy_framework"]
```

Comment (spyrostz) [reply d3a_interface/__init__.py:5]: For backwards compatibility. The main rationale behind this is for the users that have setup scripts to still continue to use them with the old imports.

**Review by fievelk — COMMENTED** [review]
There are some mentions of d3a-interface and d3a in the following files. Should we deal with them in the context of this PR, or should we do it later?
- README.md
- [d3a-interface_ci.yaml](https://github.com/gridsingularity/d3a-interface/blob/1026e83645f364a3dfdd0dbac5d2aac4aa2f6e73/.github/workflows/d3a-interface_ci.yaml)
- [d3a-interface_ci_cd_development.yaml](https://github.com/gridsingularity/d3a-interface/blob/1026e83645f364a3dfdd0dbac5d2aac4aa2f6e73/.github/workflows/d3a-interface_ci_cd_development.yaml)
- [d3a-interface_ci_cd_staging.yaml](https://github.com/gridsingularity/d3a-interface/blob/1026e83645f364a3dfdd0dbac5d2aac4aa2f6e73/.github/workflows/d3a-interface_ci_cd_staging.yaml)

**Comment by faizan2590** [comment]
> There are some mentions of d3a-interface and d3a in the following files. Should we deal with them in the context of this PR, or should we do it later?
> 
> * README.md
> * [d3a-interface_ci.yaml](https://github.com/gridsingularity/d3a-interface/blob/1026e83645f364a3dfdd0dbac5d2aac4aa2f6e73/.github/workflows/d3a-interface_ci.yaml)
> * [d3a-interface_ci_cd_development.yaml](https://github.com/gridsingularity/d3a-interface/blob/1026e83645f364a3dfdd0dbac5d2aac4aa2f6e73/.github/workflows/d3a-interface_ci_cd_development.yaml)
> * [d3a-interface_ci_cd_staging.yaml](https://github.com/gridsingularity/d3a-interface/blob/1026e83645f364a3dfdd0dbac5d2aac4aa2f6e73/.github/workflows/d3a-interface_ci_cd_staging.yaml)

To be dealt later as part of follow-up ticket to adapt all ci_cd scripts

**Review by fievelk — APPROVED** [review]

**Pull request merged**